drivers/serial/uart_nrfx_uarte.c

```diff
@@ -0,0 +1,612 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief Driver for Nordic Semiconductor nRF52 UARTE
+ */
+
+#include <uart.h>
+#include <gpio.h>
+#include <hal/nrf_uarte.h>
+#include <hal/nrf_gpio.h>
+
+/* Maximum allowed bufer size for Easy DMA transfer */
+#define UART_MAX_TX_BUFF_SIZE		BIT_MASK(UARTE0_EASYDMA_MAXCNT_SIZE)
```

Comment:
This should be calculated per instance. What about something like this:
```
#define UARTE_TX_BUFFER_SIZE(idx) \
	CONFIG_UART_##idx##_NRF_TX_BUFFER_SIZE < \
		BIT_MASK(UARTE##idx##_EASYDMA_MAXCNT_SIZE) ? \
	CONFIG_UART_##idx##_NRF_TX_BUFFER_SIZE : \
		BIT_MASK(UARTE##idx##_EASYDMA_MAXCNT_SIZE)
```
and then just:
```
static u8_t uarte##idx##_tx_buffer[UARTE_TX_BUFFER_SIZE(idx)];
```
And all this buffer size stuff is no longer needed here. It would be better to move it closer to the `UART_NRF_UARTE_DEVICE` macro where it is actually used.